### PR TITLE
! Unify errrors in the top-level posting functions a bit and offer mods ...

### PR DESCRIPTION
**Pull request by live627**
This commit simplifies posting errors a bit so that they may be easily extensible. Also allows for more extensive errors - using sprintf(). 

Again, I neglected to sign the commit off, so doing it here:
Signed-off-by: John Rayes live627@gmail.com

### CODE REVIEW

**Comment by norv** [comment]
With git, you can always go back on a commit you made, and sign it off, by running:
git commit --amend -s
(if it was the last), or respectively specify -C commit_hash.

I would highly prefer us to get used to signing off commits. It is a requirement for smCore/SMF 3.0.

**Comment by live627** [comment]
GitHub for Windows doesn't have an automated sign off feature that I could find... so I must remember to do it manually.

**Comment by norv** [comment]
If it allows amend of a commit (and it should), simply the operation of amending a commit, allows to modify the commit message as well.

File: Sources/Post.php [hunk 1]
```diff
@@ -385,11 +382,11 @@ function Post()
...
 		{
 			if (htmltrim__recursive(htmlspecialchars__recursive($_REQUEST['subject'])) == '')
-				$context['post_error']['no_subject'] = true;
+				$post_errors[] = 'no_subject';
 			if (htmltrim__recursive(htmlspecialchars__recursive($_REQUEST['message'])) == '')
-				$context['post_error']['no_message'] = true;
+				$post_errors[] = 'no_message';
 			if (!empty($modSettings['max_messageLength']) && $smcFunc['strlen']($_REQUEST['message']) > $modSettings['max_messageLength'])
-				$context['post_error']['long_message'] = true;
+				$post_errors[] = 'long_message';
```
Comment (emanuele45) [inline Sources/Post.php]: Why not:

```
$post_errors[] = array('long_message', $modSettings['max_messageLength']);
```

so you wouldn't even need the exception below:

```
if ($post_error == 'long_message')
```

Comment (live627) [inline Sources/Post.php]: Hey, good idea. I'll get to this tomorrow.

**Comment by Spuds** [comment]
Looks like an array_unique($post_errors) may be needed, since the old code could set the same message in more that one place, 

e.g. Doing a preview on a blank subject will set the 'no subject' error twice, not a problem when done as $context['post_error']['no_subject'] but when done as $post_error[] = 'no_subject' it will show up.  

Not sure if there are other dupes though, could just be the 'no subject' and could be fixed in the logic flow  ... but easy to add array_unique($post_errors) as it only gets called once in while anyway, could help mods as well.

**Comment by live627** [comment]
Oh, yeah, it does.

However, I just got another idea  which may oor may not work. What do you think about removing error processing from Post() and having it all in Post2()?

**Comment by Spuds** [comment]
Thats an interesting idea, not sure how much work would be involved, but certainly having all the error checking in one place is a more proper thing to do.  Might also be able to move everything to $post_errors instead of a combo of that and $context['post_error']['messages'][]  ... Perhaps its own function like Post_Validate() or something would give more flexibility.  Anyway if you are willing to take a look at whats involved I believe its a good update to consider.

**Pull request merged**